server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1582,52 +1663,40 @@ private void executeOneStaleIndexDelete(
                                 if (snapshotUUID.isPresent()) {
-                                    RemoteStoreShardShallowCopySnapshot remoteStoreShardShallowCopySnapshot =
-                                        REMOTE_STORE_SHARD_SHALLOW_COPY_SNAPSHOT_FORMAT.read(
-                                            shardBlob.getValue(),
-                                            snapshotUUID.get(),
-                                            namedXContentRegistry
-                                        );
-                                    String indexUUID = remoteStoreShardShallowCopySnapshot.getIndexUUID();
-                                    String remoteStoreRepoForIndex = remoteStoreShardShallowCopySnapshot.getRemoteStoreRepository();
-                                    // Releasing lock files before deleting the shallow-snap-UUID file because in case of any failure
-                                    // while releasing the lock file, we would still have the corresponding shallow-snap-UUID file
-                                    // and that would be used during next delete operation for releasing this stale lock file
-                                    RemoteStoreLockManager remoteStoreMetadataLockManager = remoteStoreLockManagerFactory.newLockManager(
-                                        remoteStoreRepoForIndex,
-                                        indexUUID,
-                                        shardBlob.getKey()
-                                    );
-                                    remoteStoreMetadataLockManager.release(
-                                        FileLockInfo.getLockInfoBuilder().withAcquirerId(snapshotUUID.get()).build()
-                                    );
-                                    if (!isIndexPresent(clusterService, indexUUID)) {
-                                        // this is a temporary solution where snapshot deletion triggers remote store side
-                                        // cleanup if index is already deleted. We will add a poller in future to take
-                                        // care of remote store side cleanup.
-                                        // see https://github.com/opensearch-project/OpenSearch/issues/8469
-                                        new RemoteSegmentStoreDirectoryFactory(
-                                            remoteStoreLockManagerFactory.getRepositoriesService(),
-                                            threadPool
-                                        ).newDirectory(
-                                            remoteStoreRepoForIndex,
-                                            indexUUID,
-                                            new ShardId(Index.UNKNOWN_INDEX_NAME, indexUUID, Integer.parseInt(shardBlob.getKey()))
-                                        ).close();
+                                    if (!releaseRemoteStoreLockAndCleanup(
```

Comment:
Wondering in cases when RemoteCleanup is throwing the `RejectedExecutionException`, should be just bail out of the index cleanup ?